pongObjects_Methods.js

```diff
@@ -1,175 +1,190 @@
-};
-//draw/Render all objects onto canvas
-function render(){
-    drawPaddle(0, 0, cvs.width, cvs.height, "BLACK");
-    drawTextscore(cvs.width/4, cvs.height/5, "WHITE", userPad.score); //score for user
-    drawTextscore(2*cvs.width/4 + 125, cvs.height/5, "WHITE", aiPad.score); //score for aiPad
-    drawPaddle(userPad.x, userPad.y, userPad.width, userPad.height, userPad.color); //draw rectangle paddle for user
-    drawPaddle(aiPad.x, aiPad.y, aiPad.width, aiPad.height, aiPad.color); //draw rectangle paddle for AI
-    drawCircle(ball.x, ball.y, ball.radius, ball.color);
-    drawNet();
-}
-//change of X direction
-function collision(ball, paddle){
-    ball.top = ball.y - ball.radius;
-    ball.bottom = ball.y + ball.radius;
-    ball.left = ball.x - ball.radius;
-    ball.right = ball.x + ball.radius; //4 points of contact is measured at the top, bottom, left, right (there is no contact at an angle)
-
-    paddle.top = paddle.y
-    paddle.bottom = paddle.y + paddle.height;
-    paddle.left = paddle.x;
-    paddle.right = paddle.x + paddle.width;
-
-    return ball.right > paddle.left && ball.bottom > paddle.top && ball.left < paddle.right && ball.top < paddle.bottom; //return true is all conditions are met
-}
-function reset(){
-    ball.x = cvs.width/2;
-    ball.y = cvs.height/2;
-    ball.speed = 5;
-// we only need to change the velocityX direction, i.e if user loses it means initially the ball was moving left 
-// so we start the ball moving right, vice versa
-    ball.velocityX *= -1
-}
-function updateBallPos(){
-    ball.x += ball.velocityX; //Increments the ball x,y direction by 5 or by velocityX/Y; This will send the ball towards the bottom right 
-    ball.y += ball.velocityY; //Think of bottom right as our +X, +Y 
-}
-//if bottom of ball (bally + ballrad) > heightofcanvas or if top of ball(bally - ballradius) < 0(top of canvas) 
-//change of Y direction
-function ballCanvascollision(){
-    if(ball.y + ball.radius > cvs.height || ball.y - ball.radius < 0){ 
-        ball.velocityY = -(ball.velocityY); 
+
+    drawCanvas(){
+        this._ctx.fillStyle = "BLACK";
+        this._ctx.fillRect(0,0,this._cvs.width, this._cvs.height);
     }
-}
-function ballPaddlecollision(){
-    //checks which paddle was hit ? userPaddle or aiPaddle
-    let paddlehit = (ball.x < cvs.width/2) ? userPad : aiPad; 
-    //if collision return True, its a collision on the paddle and X direction according to (speed, angle, direction and velocity)
-    if(collision(ball, paddlehit)){
-        let collidePoint = ball.y - (paddlehit.y + paddlehit.height/2);
 
-        collidePoint = collidePoint/(paddlehit.height/2);
+    drawNet(){
+        this._ctx.fillStyle = "WHITE";
+        this._ctx.fillRect(299, 0, 2, 400);
+    }
 
-        let angleRad = collidePoint * Math.PI/4;
+    aiTracking() {
+        this._ai._y += (this._ball._y - (this._ai._y + (this._ai._height/2))) * this._aiDifficulty;
```

Comment:
actual OOP with private variables don't actually call the variable itself from outer classes. instead, they have getter and setter methods. in this instance, you'd have a setter method in your Paddle class called
`setYPos(newYPos) {
    this._y = newYPos;
}`